import-scripts/transfer-deployment-color.sh

```diff
@@ -0,0 +1,917 @@
+}
+
+function read_array() {
+    local key="$1"
+    local -n target_ref=$2
+    local type
+    type=$("$YQ_BINARY" eval -r "$key | type" "$CONFIG_FILEPATH")
+    if [ "$type" != "!!seq" ] ; then
+        echo "Error : expected array at '$key' in $CONFIG_FILEPATH" >&2
+        exit 1
+    fi
+    mapfile -t target_ref < <("$YQ_BINARY" eval -r "$key[]" "$CONFIG_FILEPATH")
+    if [ "${#target_ref[@]}" -eq 0 ] ; then
+        echo "Error : array at '$key' in $CONFIG_FILEPATH must contain at least one value" >&2
+        exit 1
+    fi
+}
+
+function read_map() {
+    local key="$1"
+    local -n target_ref=$2
+    local type
+    type=$("$YQ_BINARY" eval -r "$key | type" "$CONFIG_FILEPATH")
+    if [ "$type" != "!!map" ] ; then
+        echo "Error : expected map at '$key' in $CONFIG_FILEPATH" >&2
+        exit 1
+    fi
+    target_ref=()
+    while IFS=$'\t' read -r entry_key entry_value ; do
+        target_ref["$entry_key"]="$entry_value"
+    done < <("$YQ_BINARY" eval -r "$key | to_entries[] | \"\(.key)\t\(.value)\"" "$CONFIG_FILEPATH")
+    if [ "${#target_ref[@]}" -eq 0 ] ; then
+        echo "Error : map at '$key' in $CONFIG_FILEPATH must contain at least one entry" >&2
+        exit 1
+    fi
+}
+
+function load_configuration() {
+    local config_path="$1"
+    if ! [ -f "$config_path" ] || ! [ -r "$config_path" ] ; then
+        echo "Error : unable to read config file '$config_path'" >&2
+        exit 1
+    fi
+
+    SERVICE_ACCOUNT=$(read_scalar '.service_account')
+    CLUSTER_KUBECONFIG=$(read_scalar '.cluster_cfg')
+    MANAGE_DATABASE_TOOL_PROPERTIES_FILEPATH=$(read_scalar '.manage_database_tool_properties_filepath')
+    tmp=$(read_scalar '.tmp')
+    CACHE_WARMING_POD_LIST_FILEPATH=$(read_scalar '.cache_warming_pod_list_filepath')
+    CACHE_WARMING_POD_SUBLIST_FILEPATH=$(read_scalar '.cache_warming_pod_sublist_filepath')
+    KS_K8S_DEPL_REPO_DIRPATH=$(read_scalar '.ks_k8s_depl_repo_dirpath')
+    INGRESS_YAML_FILEPATH=$(read_scalar '.ingress_yaml_filepath')
+    REPLICA_READY_CHECK_PAUSE_SECONDS=$(read_scalar '.replica_ready_check_pause_seconds')
+    REPLICA_READY_CHECK_MAX_CHECKCOUNT=$(read_scalar '.replica_ready_check_max_checkcount')
+    GIT_BINARY=$(read_scalar '.git_binary')
+    WARM_CACHES=$(read_scalar '.warm_caches')
+    SYNCHRONIZE_USER_TABLES=$(read_scalar '.synchronize_user_tables')
+    CLEAR_PERSISTENCE_CACHES_BLUE_FUNCTION=$(read_scalar '.clear_persistence_caches_blue_function')
+    CLEAR_PERSISTENCE_CACHES_GREEN_FUNCTION=$(read_scalar '.clear_persistence_caches_green_function')
+
+    read_array '.blue_deployment_list' BLUE_DEPLOYMENT_LIST
+    read_array '.green_deployment_list' GREEN_DEPLOYMENT_LIST
+    read_map '.deployment_to_full_replica_count_map' DEPLOYMENT_TO_FULL_REPLICA_COUNT_MAP
+    read_map '.deployment_to_cache_warming_policy' DEPLOYMENT_TO_CACHE_WARMING_POLICY
+    read_map '.deployment_to_yaml_filepath_map' DEPLOYMENT_TO_YAML_FILEPATH_MAP
+}
+
+function usage() {
```

Comment:
I usually like to keep the usage() function up near the top of the script so that when the script is opened it can be easily referenced up top ... functioning somewhat like file-level documentation.